test/e2e/workflow/deployment_test.go

```diff
@@ -454,6 +455,124 @@ var _ = Describe("NetworkAddonsConfig", func() {
+					WithTransform(getDaemonsetTemplateLabels, HaveKeyWithValue(names.COMPONENT_LABEL_KEY, componentLabelCRlVal))), "label and template label values of the daemonset should be contain the values on NetworkAddonsConfig")
+			})
+
+			It("Should take the `part-of`, `version` and `component` labels value and apply it to the deployed deployments", func() {
+				Eventually(func() *v1.Deployment {
+					err := framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: KubeMacPoolComponent.Deployments[0], Namespace: components.Namespace}, kubemacpoolDeployment)
+					Expect(err).ToNot(HaveOccurred(), "should succeed getting the kubemacpool deployment")
+					return kubemacpoolDeployment
+				}, 2*time.Minute, time.Minute).Should(SatisfyAll(
+					WithTransform(getDeploymentLabels, HaveKeyWithValue(names.PART_OF_LABEL_KEY, partOfLabelCRVal)),
+					WithTransform(getDeploymentLabels, HaveKeyWithValue(names.VERSION_LABEL_KEY, versionLabelCRVal)),
+					WithTransform(getDeploymentLabels, HaveKeyWithValue(names.COMPONENT_LABEL_KEY, componentLabelCRlVal)),
+					WithTransform(getDeploymentTemplateLabels, HaveKeyWithValue(names.PART_OF_LABEL_KEY, partOfLabelCRVal)),
+					WithTransform(getDeploymentTemplateLabels, HaveKeyWithValue(names.VERSION_LABEL_KEY, versionLabelCRVal)),
+					WithTransform(getDeploymentTemplateLabels, HaveKeyWithValue(names.COMPONENT_LABEL_KEY, componentLabelCRlVal))), "label and template label values of the deployment should be contain the values on NetworkAddonsConfig")
+			})
+		})
+
+		Context("When NetworkAddonsConfig is labeled with `managed-by` label", func() {
+			const (
+				managedByLabelCRVal        = "managed-by-cr-value"
+				managedByLabelComponentVal = names.MANAGED_BY_LABEL_DEFAULT_VALUE
+			)
+			labels := map[string]string{names.MANAGED_BY_LABEL_KEY: managedByLabelCRVal}
+
+			BeforeEach(func() {
+				Eventually(func() *v1.DaemonSet {
+					err := framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: OvsComponent.DaemonSets[0], Namespace: components.Namespace}, ovsDaemonSet)
+					Expect(err).ToNot(HaveOccurred(), "should succeed getting the ovs daemonSet")
+					return ovsDaemonSet
+				}, 2*time.Minute, time.Minute).Should(SatisfyAll(
+					WithTransform(getDaemonsetLabels, HaveKeyWithValue(names.MANAGED_BY_LABEL_KEY, managedByLabelComponentVal)),
+					WithTransform(getDaemonsetTemplateLabels, HaveKeyWithValue(names.MANAGED_BY_LABEL_KEY, managedByLabelComponentVal))),
+					"Daemonset should have been labeled with `managed-by` label with value %s", managedByLabelComponentVal)
+
+				LabelConfig(gvk, labels)
+				Eventually(func() map[string]string {
+					return GetConfig(gvk).GetLabels()
+				}, 2*time.Minute, time.Minute).Should(HaveKeyWithValue(names.MANAGED_BY_LABEL_KEY, managedByLabelCRVal),"should succeed setting labels on NetworkAddonsConfig")
+			})
+
+			It("Should NOT apply `managed-by` label value to the deployed daemonSet", func() {
```

Comment:
I think its better to speak in the "positive" terms
Should keep the original managed-by label value
(it is too much oriented to the bug that was, and when time goes by what is important is the current logic)